cmd/notary/integration_test.go

```diff
@@ -1526,3 +1532,374 @@ func TestWitness(t *testing.T) {
+
+	// list delegations - we should see our one delegation
+	output, err = runCommand(t, tempDir, "-s", server.URL, "delegation", "list", "gun")
+	require.NoError(t, err)
+	require.NotContains(t, output, "No delegations present in this repository.")
+
+	// remove the targets key to demonstrate that delegates don't need this key
+	keyDir := filepath.Join(tempDir, "private", "tuf_keys")
+	require.NoError(t, os.Remove(filepath.Join(keyDir, "gun", targetKeyID+".key")))
+
+	// we are now set up with the first part, now import the delegation key- add a target- publish
+
+	// first test the negative case, should fail without the key import
+
+	// add a target using the delegation -- will only add to targets/releases
+	_, err = runCommand(t, tempDir, "add", "gun", target, tempTargetFile.Name(), "--roles", "targets/releases")
+	require.NoError(t, err)
+
+	// list targets for targets/releases - we should see no targets until we publish
+	output, err = runCommand(t, tempDir, "-s", server.URL, "list", "gun", "--roles", "targets/releases")
+	require.NoError(t, err)
+	require.Contains(t, output, "No targets")
+
+	output, err = runCommand(t, tempDir, "-s", server.URL, "status", "gun")
+	require.NoError(t, err)
+
+	// publish repo
+	_, err = runCommand(t, tempDir, "-s", server.URL, "publish", "gun")
+	require.Error(t, err)
+
+	// list targets for targets/releases - we should not see our target!
+	output, err = runCommand(t, tempDir, "-s", server.URL, "list", "gun", "--roles", "targets/releases")
+	require.NoError(t, err)
+	require.Contains(t, output, "No targets present")
+
+	// now test for the positive case, import the key and publish and it should work
+
+	// the changelist still exists so no need to add the target again
+	// just import the key and publish
+	output, err = runCommand(t, tempDir, "-s", server.URL, "status", "gun")
+	require.NoError(t, err)
+	require.Contains(t, output, "targets/releases")
+	require.Contains(t, output, "sdgkadga")
+
+	// import the key
+	_, err = runCommand(t, tempDir, "key", "import", keyFile.Name(), "-r", "targets/releases", "-g", "gun")
+	require.NoError(t, err)
+
+	// make sure that it has been imported fine
+	// if there is hardware available, root will only be on hardware, and not
+	// on disk
+	_, err = os.Open(filepath.Join(tempDir, "private", notary.NonRootKeysSubdir, privKey.ID()+".key"))
+	require.NoError(t, err)
+
+	// now try to publish
+	_, err = runCommand(t, tempDir, "-s", server.URL, "publish", "gun")
+	require.NoError(t, err)
+
+	// check that changelist is applied
+	output, err = runCommand(t, tempDir, "-s", server.URL, "status", "gun")
+	require.NoError(t, err)
+	require.NotContains(t, output, "targets/releases")
+
+	// list targets for targets/releases - we should see our target!
+	output, err = runCommand(t, tempDir, "-s", server.URL, "list", "gun", "--roles", "targets/releases")
+	require.NoError(t, err)
+	require.NotContains(t, output, "No targets present")
+	require.Contains(t, output, "sdgkadga")
```

Comment:
nit: can we also check that `targets/releases` is in the output?  Just to confirm that we published under that role and not another one